sdk/tables/azure-data-tables/inc/azure/data/tables/table_audience.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#pragma once
+
+#include <string>
+#include <utility>
+
+namespace Azure { namespace Data { namespace Tables {
+  /**
+   * @brief Audiences available for Azure Table storage service.
+
+   *
+   */
+  class TableAudience final {
+    std::string m_audience;
+
+  public:
+    /**
+     * @brief Get the audience value.
+     */
+    std::string GetAudience() const { return m_audience; }
+    /**
+     * @brief Construct a new TablesAudience object
+     *
+     * @param tableAudience The Azure Active Directory audience to use when forming authorization
+     * scopes. For the Language service, this value corresponds to a URL that identifies the Azure
+     * cloud where the resource is located. For more information: See
+     * https://learn.microsoft.com/azure/storage/tables/authorize-access-azure-active-directory
+     */
+    explicit TableAudience(std::string tableAudience) : m_audience{std::move(tableAudience)} {}
+
+    /**
+     * @brief The service endpoint for a given storage account. Use this method to acquire a token
+     * for authorizing requests to that specific Azure Storage account and service only.
+     *
+     * @param tablesAccountName The storage account name used to populate the service endpoint.
+     * @return The service endpoint for a given storage account.
+     */
+    static TableAudience CreateTablesServiceAccountAudience(const std::string& tablesAccountName)

```

Comment:
```suggestion
    static TableAudience CreateTableServiceAccountAudience(const std::string& tablesAccountName)

```
also, where did you get the name from?

Reply:
i.e. why is it not "CreateAccountAudience()"